llvm-build/utils.ps1

```diff
@@ -0,0 +1,278 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+if (Test-Path function:\exec) {
+    # already included
+    return
+}
+
+if (!(Test-Path function:\Get-RepoRoot)) {
+    # git revparse uses cwd. E2E builds use a different
+    # working dir, so we pin it to out repo (submodule in E2E)
+    function Get-RepoRoot {
+        exec -wd $PSScriptRoot {
+            git rev-parse --show-toplevel
+        }
+    }
+}
+
+# Fix temp path for non-windows platforms if missing
+if (!(Test-Path env:\TEMP)) {
+    $env:TEMP = [System.IO.Path]::GetTempPath()
+}
+
+# Test whether we build the compiler components.
+function Test-BuildCompiler {
+    # By default we want to build the compiler
+    # If no env var is defined we default to true
+    if (!(Test-Path env:\ENABLE_COMPILER)) {
+        $true
+    }
+    else {
+        # The env var exists, we use its value
+        # Coerce falsy values and negate to keep the defalt $true
+        $env:ENABLE_COMPILER -ne $false
+    }
+}
```

Comment:
(I don't know powershell)
I have an impression that the lines 29 and 34 return a value. And if someone adds some code between lines 35 and 36 then the value returned can become unexpected. 
If supported, I would consider an explicit `return` inside of `if` and skipping `else` after `return`:
```suggestion
function Test-BuildCompiler {
    # By default we want to build the compiler
    # If no env var is defined we default to true
    if (!(Test-Path env:\ENABLE_COMPILER)) {
        return $true
    }
    # The env var exists, we use its value
    # Coerce falsy values and negate to keep the defalt $true
    return ( $env:ENABLE_COMPILER -ne $false )
}
```

Reply:
PowerShell allows for multiple returns from a function, so if any code was added after this it might cause the function to return a list of values instead of just one. That would still break callers, as you point out, and potentially be confusing to root cause. All that said, as I mentioned above it seems several of these utils aren't actually used, so I think this whole function can just be removed.